### ISSUE TITLE
nx release command fails with projects made with nx/node that doesn't have a package.json file inside the project dir

### ISSUE THREAD

**Issue by ctrlmaniac** [nrwl/nx#22363]
### Current Behavior

nx release fails when package.json is missing in projects

### Expected Behavior

nx release to run even without a package.json or to use the package.json inside the dist folder

### Steps to Reproduce

1. `nx g @nx/node:app api`
2. `nx release --dry-run`

### Nx Report

```shell
Node   : 20.11.1
OS     : linux-x64
npm    : 10.2.4

nx (global)        : 18.0.8
nx                 : 18.0.7
@nx/js             : 18.0.7
@nx/jest           : 18.0.7
@nx/linter         : 18.0.7
@nx/eslint         : 18.0.7
@nx/workspace      : 18.0.7
@nx/devkit         : 18.0.7
@nx/eslint-plugin  : 18.0.7
@nx/express        : 18.0.7
@nx/node           : 18.0.7
@nx/plugin         : 18.0.7
@nx/react          : 18.0.7
@nrwl/tao          : 18.0.7
@nx/vite           : 18.0.7
@nx/web            : 18.0.7
@nx/webpack        : 18.0.7
typescript         : 5.3.3
```

### Failure Logs

```shell
NX   The project "api" does not have a package.json available at apps/api/package.json.

To fix this you will either need to add a package.json file at that location, or configure "release" within your nx.json to exclude "api" from the current release group, or amend the packageRoot configuration to point to where the package.json should be.
```

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by FrozenPandaz** [comment]
Could you please provide a reproduction of this issue so we can see more about what you are trying to do?

If you want to publish your node api to NPM, then you'll need a `package.json`. 

But it seems more like you want to deploy your Node API to be hosted somewhere.

**Comment by ctrlmaniac** [comment]
Yes indeed! I need to deploy my application. It needs to be hosted on a VPS. I've found it to be a little tricky! Firstly I can't release it because it's not a js library, secondly I needed to manually create a package.json file inside the src, then the build target will pick it and add the dependencies manually. In this way the application can be deployed

**Comment by fahslaj** [comment]
Hi @ctrlmaniac and @mpsanchis, the release version generator is currently designed to manage versions of npm packages, which require a `package.json` file. You're right in that in the context of deploying applications, much less should be required. Applications don't need to use the `semver` pattern for versioning, don't need a `package.json` file, and many version generator options wouldn't be relevant. I believe the best way to support this type of application deployment would be through a different version generator implementation.

While it's currently _possible_ to replace the version generator, we recognize that it's difficult because of the sheer amount of functionality in the existing `@nx/js:release-version` implementation. We've got plans to extract utility functions and export them from Nx directly to make writing your own version generators easier, but we're just not quite there yet.

We appreciate your input. I'm going to convert this to a feature so that it can be prioritized accordingly.
